src/com/amplitude/api/Diagnostics.java

```diff
@@ -0,0 +1,187 @@
+package com.amplitude.api;
+
+import android.text.TextUtils;
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.FormBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Created by djih on 11/1/18.
+ */
+
+public class Diagnostics {
+
+    public static final String DIAGNOSTIC_EVENT_ENDPOINT = "https://api.amplitude.com/diagnostic";
+
+    public static final int DIAGNOSTIC_EVENT_API_VERSION = 1;
+
+    public static final int DIAGNOSTIC_EVENT_MAX_COUNT = 50; // limit memory footprint
```

Comment:
we would have to build that into the event server probably. currently the only way I limit that is via this max event count, so it would allow like at most 50 per 30 seconds

Reply:
I think it might be more useful for us to collect unique errors, so maybe within a batch of 50, ensure that we only have 1 of a certain type of error string? Thoughts @curtisliu

Reply:
maybe unique errors and number of occurrences?